server/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSourceType.java

```diff
@@ -98,4 +98,10 @@ ValuesSource replaceMissing(ValuesSource valuesSource, Object rawMissing, DocVal
     default DocValueFormat getFormatter(String format, ZoneId tz) {
         return DocValueFormat.RAW;
     }
+
+    /**
+     * Returns the name of the Values Source Type for stats purposes
+     * @return the name of the Values Source Type
+     */
+    String typeName();
```

Comment:
This name is used only under the aggregation name so "foo" in aggregation "bar" can be different from "foo" in the aggregation  "baz"  and it is perfectly fine because this "foo" will be "assessed" only in the context of aggregation usage. In usage, we are trying to use user-friendly names. So fully qualified class names will be very out of place there.